### PR TITLE
[5.5] Migrate drop command

### PR DESCRIPTION
**Pull request by lkmadushan**
This extracts drop tables into a separate command and used drop command instead of rollback in DatabaseMigrations trait to make tests run quickly. Also refactored migrate:fresh command to use use migrate:drop command

### CODE REVIEW

File: src/Illuminate/Foundation/Providers/ArtisanServiceProvider.php [hunk 1]
```diff
@@ -528,6 +530,18 @@ protected function registerMigrateRollbackCommand()
      *
      * @return void
      */
+    protected function registerMigrateDropCommand()
+    {
+        $this->app->singleton('command.migrate.rollback', function ($app) {
```
Comment (m1guelpf) [inline src/Illuminate/Foundation/Providers/ArtisanServiceProvider.php]: The singleton should be 'command.migrate.drop'

**Review by m1guelpf — CHANGES_REQUESTED** [review]

**Comment by srmklive** [comment]
No need for it in my opinion. You can accomplish the same task with `migrate:reset`.

**Comment by lkmadushan** [comment]
@srmklive This can be useful if someone wants to drop all tables at once without going through any UI. migrate:reset could throw errors some situations.

**Comment by browner12** [comment]
can you explain the use case for dropping all the tables, but *not* migrating after that?

**Comment by lkmadushan** [comment]
@browner12 no idea. but hope this will be helpful for DatabaseMigrations when testing

**Comment by taylorotwell** [comment]
No plans to add at this time.

**Pull request closed without merging**